library/Icinga/Web/Navigation/ConfigMenu.php

```diff
@@ -0,0 +1,293 @@
+<?php
+
+namespace Icinga\Web\Navigation;
+
+use Icinga\Application\Hook\HealthHook;
+use Icinga\Application\Icinga;
+use Icinga\Authentication\Auth;
+use ipl\Html\Attribute;
+use ipl\Html\Attributes;
+use ipl\Html\BaseHtmlElement;
+use ipl\Html\HtmlElement;
+use ipl\Html\HtmlString;
+use ipl\Html\Text;
+use ipl\Web\Url;
+use ipl\Web\Widget\Icon;
+use ipl\Web\Widget\StateBadge;
+
+class ConfigMenu extends BaseHtmlElement
+{
+    const STATE_OK = 'ok';
+    const STATE_CRITICAL = 'critical';
+    const STATE_WARNING = 'warning';
+    const STATE_PENDING = 'pending';
+    const STATE_UNKNOWN = 'unknown';
+
+    protected $tag = 'ul';
+
+    protected $defaultAttributes = [
+        'class' => 'nav nav-level-1'
+    ];
+
+    protected $children = [
+        'system' => [
+            'title' => 'System',
+            'items' => [
+                'about' => [
+                    'label' => 'About',
+                    'url' => 'about'
+                ],
+                'health' => [
+                    'label' => 'Health',
+                    'url' => 'health',
+                ],
+                'announcements' => [
+                    'label' => 'Announcements',
+                    'url' => 'announcements'
+                ],
+                'sessions' => [
+                    'label' => 'User Sessions',
+                    'permission'  => 'application/sessions',
+                    'url'         => 'manage-user-devices'
+                ]
+            ]
+        ],
+        'configuration' => [
+            'title' => 'Configuration',
+            'permission' => 'config/*',
+            'items' => [
+                'application' => [
+                    'label' => 'Application',
+                    'url' => 'config/general'
+                ],
+                'authentication' => [
+                    'label' => 'Access Control',
+                    'permission'  => 'config/access-control/*',
+                    'url' => 'role/list'
+                ],
+                'navigation' => [
+                    'label' => 'Shared Navigation',
+                    'permission'  => 'config/navigation',
+                    'url' => 'navigation'
+                ],
+                'modules' => [
+                    'label' => 'Modules',
+                    'permission'  => 'config/modules',
+                    'url' => 'config/modules'
+                ]
+            ]
+        ],
+        'logout' => [
+            'items' => [
+                'logout' => [
+                    'label'       => 'Logout',
+                    'atts'  => [
+                        'target' => '_self',
+                        'class' => 'nav-item-logout'
+                    ],
+                    'url'         => 'authentication/logout'
+                ]
+            ]
+        ]
+    ];
+
+    protected $selected;
+
+    protected $cogItemActive = false;
+
+    protected function assembleUserMenuItem(HtmlElement $userMenuItem)
```

Comment:
Don't use `HtmlElement` as type here. It should never be used this way, as it's just a factory. Use `BaseHtmlElement` instead.

Reply:
✅

Reply:
✅